File: day4/day4.rs
use std::fs::File;
use std::io::{self, prelude::*, BufReader};
use regex::Regex;

fn main() -> io::Result<()> {
    let file = File::open("./input.txt")?;
    let reader = BufReader::new(file);
    let mut overlaps = 0;

    for line in reader.lines() {
      let re = Regex::new("[-,]");
      let row = line.unwrap();
      let strs = re.unwrap().split(&row).collect::<Vec<&str>>();
      let nums = strs.iter().map(|&x| x.to_string().parse::<i32>().unwrap()).collect::<Vec::<i32>>();

      // if (nums[0] <= nums[2] && nums[1] >= nums[3]) || (nums[2] <= nums[0] && nums[3] >= nums[1]) {
      if (nums[0] <= nums[2] && nums[2] <= nums[1]) || (nums[0] <= nums[3] && nums[3] <= nums[1]) || (nums[0] <= nums[2] && nums[1] >= nums[3]) || (nums[2] <= nums[0] && nums[3] >= nums[1]) {
        overlaps += 1;
      }
    }

    println!("{}", overlaps);
    Ok(())
}

File: day4/day4.kt
package one

import java.io.File

fun parseLine(line: String): List<Int> {
    return line.split(Regex("[-,]")).map { x -> Integer.parseInt(x) }
}

fun main() {
    var lines = File("input.txt").readLines().map { x -> parseLine(x) }
    var count = 0
    for (line in lines) {
        if (line[0] <= line[2] && line[1] >= line[3]) {
            count++
        } else if (line[2] <= line[0] && line[3] >= line[1]) {
            count++
        } else if (line[0] <= line[2] && line[1] >= line[2]) {
            count++
        } else if (line[0] <= line[3] && line[1] >= line[3]) {
            count++
        }
    }
    println(count)
}
